### PR TITLE
Android sample uses the plugins {} block to apply the Android plugin

### PR DESCRIPTION
**Pull request by eskatos**
By using a plugin resolution strategy.

This enable removing gradle/project-schema.json and relying on jit
project accessors instead.

See #406

### CODE REVIEW

File: samples/hello-android/settings.gradle [hunk 1]
```diff
@@ -1 +1,15 @@
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+        maven { url "https://jcenter.bintray.com/" }
+    }
+    resolutionStrategy {
+        eachPlugin {
+            if(requested.id.id == "com.android.application") {
```
Comment (bamboo) [inline samples/hello-android/settings.gradle]: :x: Space missing between `if` and the parenthesis.

File: samples/hello-android/gradle/project-schema.json [hunk 2]
```diff
@@ -1,73 +0,0 @@
-{
```
Comment (bamboo) [inline samples/hello-android/gradle/project-schema.json]: 👍

File: samples/hello-android/build.gradle.kts [hunk 3]
```diff
@@ -1,17 +1,6 @@
...
-    }
-    repositories {
-        jcenter()
-    }
-}
-
-apply {
-    plugin("com.android.application")
-    plugin("kotlin-android")
+plugins {
```
Comment (bamboo) [inline samples/hello-android/build.gradle.kts]: Niiiiice.

File: samples/hello-android/settings.gradle (hunk 1 above)
Comment (eskatos) [reply samples/hello-android/settings.gradle]: Fixed

File: samples/hello-android/settings.gradle [hunk 4]
```diff
@@ -1 +1,15 @@
...
+    resolutionStrategy {
+        eachPlugin {
+            if (requested.id.id == "com.android.application") {
+                useModule("com.android.tools.build:gradle:${requested.version}")
+            }
+        }
+    }
+}
+
 rootProject.buildFileName = 'build.gradle.kts'
```
Comment (JLLeitschuh) [inline samples/hello-android/settings.gradle]: Is this line required anymore? Isn't this autodetected now?

Comment (bamboo) [reply samples/hello-android/settings.gradle]: @JLLeitschuh it is still required due to the existence of `build.gradle` (required by Android Studio).

Comment (JLLeitschuh) [reply samples/hello-android/settings.gradle]: Ah. Fair enough.

**Pull request merged**